test/VaultAsyncMode.spec.ts

```diff
@@ -0,0 +1,335 @@
+      });
+      await vault.connect(alice).deposit(params);
+
+      await addYieldToVault('50');
+      await vault.connect(owner).updateInvested();
+
+      await expect(
+        vault.connect(alice).claimYield(alice.address),
+      ).to.be.revertedWith('VaultNotEnoughFunds');
+    });
+
+    it("doesn't revert and withdraws funds from the Vault when yield = reserves", async () => {
+      const params = depositParams.build({
+        amount: parseUnits('100'),
+        inputToken: underlying.address,
+        claims: [claimParams.percent(100).to(alice.address).build()],
+      });
+      await vault.connect(alice).deposit(params);
+
+      await addYieldToVault('50');
+      await vault.connect(owner).updateInvested();
+
+      // bob's deposit is exactly enough to cover alice's claim and leave valut with 0 balance
+      const bobsDeposit = depositParams.build({
+        amount: parseUnits('34'),
+        inputToken: underlying.address,
+        claims: [claimParams.percent(100).to(bob.address).build()],
+      });
+      await vault.connect(bob).deposit(bobsDeposit);
+
+      await expect(vault.connect(alice).claimYield(alice.address)).to.not.be
+        .reverted;
+
+      expect(await underlying.balanceOf(vault.address)).to.eq('0');
+      expect(await underlying.balanceOf(strategy.address)).to.eq(
+        parseUnits('135'),
+      );
+    });
+
+    it("doesn't revert and withdraws funds from the Vault when yield < reserves", async () => {
+      const params = depositParams.build({
+        amount: parseUnits('100'),
+        inputToken: underlying.address,
+        claims: [claimParams.percent(100).to(alice.address).build()],
+      });
+      await vault.connect(alice).deposit(params);
+
+      await addYieldToVault('50');
+      await vault.connect(owner).updateInvested();
+
+      // bob's deposit is more than enough to cover alice's claim
+      const bobsDeposit = depositParams.build({
+        amount: parseUnits('100'),
+        inputToken: underlying.address,
+        claims: [claimParams.percent(100).to(bob.address).build()],
+      });
+      await vault.connect(bob).deposit(bobsDeposit);
+
+      await expect(vault.connect(alice).claimYield(alice.address)).to.not.be
+        .reverted;
+
+      // vault had 115 before the claim, and 49 was claimed, so it should have 115 - 49 = 66
+      expect(await underlying.balanceOf(vault.address)).to.eq(parseUnits('66'));
+      expect(await underlying.balanceOf(strategy.address)).to.eq(
+        parseUnits('135'),
+      );
+    });
+  });
+
+  describe("#withdraw doesn't rebalance the Vault's funds", () => {
+    it('reverts when amount > reserves', async () => {
+      const params = depositParams.build({
+        amount: parseUnits('100'),
+        inputToken: underlying.address,
+        claims: [claimParams.percent(100).to(alice.address).build()],
+      });
+      await vault.connect(alice).deposit(params);
+
+      await addYieldToVault('50');
```

Comment:
why do you need yield on this test and the ones that come after?

Reply:
I don't need it actually, going to remove that

Reply:
[fixed](https://github.com/lindy-labs/sc_solidity-contracts/pull/136/commits/47be529313b023a0fe04a06247a0095ceba141c9)